### PR TITLE
Force all non-copy types to return using `RETD` regardless of their size.

### LINKED ISSUE #968: Missing information in returned data when calling method that returns small arrays
I have these two methods:

```Rust
fn get_large_array() -> [u32; 2] {
    let x: [u32; 2] = [1, 2];
    x
}

fn get_single_element_array() -> [u32; 1] {
    let x: [u32; 1] = [1];
    x
}
```

The `large_array` one, as expected, returns the correct data in `ReturnData`:

```
ReturnData { id: 0xd73fc7877ec161d10d62b00cd1ec1d7ed77aaae5989e9cc8470df2111bedc3be, 
ptr: 2120, len: 16, digest: 0x8c7654ecfd7b0b623b803e2f4e02ad1cc84278efdfcd7c4c9208edd81f17e115, 
data: [0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 2], pc: 1860, is: 1600 } 
```

The `single_element_array()`, however, returns what looks like a pointer, but `Return` is the only receipt I get back. No other piece of information I could use that pointer to retrieve the `[1]` being returned:

```
Return { id: 0xd73fc7877ec161d10d62b00cd1ec1d7ed77aaae5989e9cc8470df2111bedc3be, val: 2120, pc: 1880, is: 1600 }
```

### PR DESCRIPTION
**Pull request by otrho**
Closes #968.

### CODE REVIEW

File: test/src/e2e_vm_tests/test_programs/should_pass/language/retd_small_array/Forc.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[project]
+authors = ["Fuel Labs <contact@fuel.sh>"]
+entry = "main.sw"
+license = "Apache-2.0"
+name = "retd_small_array"
+
+#[dependencies]
+#core = { git = "http://github.com/FuelLabs/sway-lib-core" }
+#std = { git = "http://github.com/FuelLabs/sway-lib-std" }
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/should_pass/language/retd_small_array/Forc.toml:9]:
```suggestion
```

File: test/src/e2e_vm_tests/test_programs/should_pass/language/retd_small_array/src/main.sw [hunk 2]
```diff
@@ -0,0 +1,6 @@
+script;
+
+fn main() -> [u32; 1] {
+    let x: [u32; 1] = [1];
```
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/should_pass/language/retd_small_array/src/main.sw:4]: Maybe also add a test for a zero-length array. Or is that not even allowed (should it be?)?

Comment (otrho) [reply test/src/e2e_vm_tests/test_programs/should_pass/language/retd_small_array/src/main.sw:4]: Good question.  Rust says empty arrays are OK so I guess Sway does too.

**Review by adlerjohn — COMMENTED** [review]
lgtm, will defer to an actual compiler engineer for approval

**Review by mohammadfawaz — APPROVED** [review]
LGTM! Thanks.

**Review by sezna — APPROVED** [review]
